src/test/java/org/sonar/plugins/checkstyle/internal/ChecksTest.java

```diff
@@ -89,8 +89,8 @@ public void verifyTestConfigurationFiles() throws Exception {
 
         Assert.assertFalse("no modules", modules.isEmpty());
 
-        validateSonarRules(new HashSet<>(modules));
-        validateSonarProperties(new HashSet<>(modules));
+        //        validateSonarRules(new HashSet<>(modules));
+        //        validateSonarProperties(new HashSet<>(modules));
```

Comment:
what is a reason ?

if this code is completely not required, please remove it completely.

Reply:
This UT checks whether check details present in `rules.xml` are consistent with actual data fetched from checks.
@romani @muhlba91 Please confirm if I shall remove completely.

Reply:
While we support old files loading , we need such test.
BUT
we need to test only single or few module loading, just to make sure we keep compatibility of loading old files.

please select few most complicated by structure/properties modules and put them such validators
`validateSonarRules(new HashSet<>(module1, module2, module3, module4));`

Reply:
But what if `module1`'s `propertyA` changes its default values? If we have to manually edit the xml file, then it will defeat the purpose

Reply:
old files will be always ignored, all checkstyle's metadata will always override whatever old metadata has.
so old files will be just for testing that we can load old files to keep compatibility.

Reply:
but in scope of this RP we need to run `validateSonarRules` on all modules that are predate checkstyle 8.36 to be sure that we can load old files. just hard code a list of them in test. We will reduce a list in next issue https://github.com/checkstyle/sonar-checkstyle/issues/321 .